run.py

```diff
@@ -50,56 +51,68 @@ def run_one_step_with_cudastreams(func, streamcount):
 
         print(f"Cuda StreamCount:{len(streamlist)}: gpu time {start_event.elapsed_time(end_event)}")
 
+
 def run_one_step(func):
 
     func()
 
     if args.device == "cuda":
-      torch.cuda.synchronize()
-      start_event = torch.cuda.Event(enable_timing=True)
-      end_event = torch.cuda.Event(enable_timing=True)
+        torch.cuda.synchronize()
+        start_event = torch.cuda.Event(enable_timing=True)
+        end_event = torch.cuda.Event(enable_timing=True)
 
-      start_event.record()
+        start_event.record()
 
-      t0 = time.time()
-      func()
-      t1 = time.time()
+        t0 = time.time()
+        func()
+        t1 = time.time()
 
-      end_event.record()
-      torch.cuda.synchronize()
-      print(f"Ran in {t1 - t0} seconds, gpu time {start_event.elapsed_time(end_event)}.")
+        end_event.record()
+        torch.cuda.synchronize()
+        print(f"Ran in {t1 - t0} seconds, gpu time {start_event.elapsed_time(end_event)}.")
 
     else:
 
-      t0 = time.time()
-      func()
-      t1 = time.time()
+        t0 = time.time()
+        func()
+        t1 = time.time()
 
-      print(f"Ran in {t1 - t0} seconds.")
+        print(f"Ran in {t1 - t0} seconds.")
 
 
 def profile_one_step(func, nwarmup=3):
     for i in range(nwarmup):
-        func()
-
-    use_cuda = args.device == "cuda"
+            func()
 
-    with profiler.profile(record_shapes=True, use_cuda = use_cuda) as prof:
+    activity_groups = []
+    if args.device == "cuda":
+        activity_groups.append(profiler.ProfilerActivity.CUDA)
+    activity_groups.append(profiler.ProfilerActivity.CPU)
+
+    with profiler.profile(
+        schedule=profiler.schedule(wait=0, warmup=0, active=1),
+        activities=activity_groups,
+        record_shapes=True,
+        on_trace_ready=profiler.tensorboard_trace_handler('./logs')
```

Comment:
shouldn't you take args.profile-folder here instead of './logs'?

Reply:
Good catch, thanks that was remains from my testing. fixing it!